packages/aws-cdk/test/api/cloudformation-deployments.test.ts

```diff
@@ -142,3 +161,751 @@ test('if toolkit stack cannot be found but SSM parameter name is present deploym
 
   expect(requestedParameterName!).toEqual('/some/parameter');
 });
+
+test('readCurrentTemplateWithNestedStacks() can handle non-Resources in the template', async () => {
+  const cfnStack = instanceMockFrom((CloudFormationStack as any));
+  CloudFormationStack.lookup = jest.fn().mockImplementation((_, stackName: string) => {
+    switch (stackName) {
+      case 'OutputParamRoot':
+        (cfnStack as any).template = jest.fn().mockReturnValue({
+          Resources: {
+            NestedStack: {
+              Type: 'AWS::CloudFormation::Stack',
+              Metadata: {
+                'aws:asset:path': 'one-output-one-param-stack.nested.template.json',
+              },
+            },
+          },
+        });
+        break;
+
+      case 'NestedStack':
+        (cfnStack as any).template = jest.fn().mockReturnValue({
+          Resources: {
+            NestedResource: {
+              Type: 'AWS::Something',
+              Properties: {
+                Property: 'old-value',
+              },
+            },
+          },
+          Parameters: {
+            NestedParam: {
+              Type: 'String',
+            },
+          },
+          Outputs: {
+            NestedOutput: {
+              Value: {
+                Ref: 'NestedResource',
+              },
+            },
+          },
+        });
+        break;
+
+      default:
+        throw new Error('unknown stack name ' + stackName + ' found in cloudformation-deployments.test.ts');
+    }
+
+    return cfnStack;
+  });
+
+  const rootStack = testStack({
+    stackName: 'OutputParamRoot',
```

Comment:
What's this `'OutputParamRoot'` name? I'm not sure what is it supposed to signify here...?

Reply:
It's supposed to signify that the stack has both and `Output` and a `Parameter`.

Reply:
Then you need to try again, because it's failing at that job right now 🙂.

How about `'ParentStackWithOutputAndParameter'`?

Reply:
+1.